core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt

```diff
@@ -0,0 +1,31 @@
+package fr.sncf.osrd.sim_infra.api
+
+import fr.sncf.osrd.railjson.schema.geom.LineString
+import fr.sncf.osrd.utils.indexing.DirStaticIdxList
+import fr.sncf.osrd.utils.indexing.StaticIdx
+import fr.sncf.osrd.utils.units.Distance
+import fr.sncf.osrd.utils.DistanceRangeMap
+
+data class Path(
+    val chunks: DirStaticIdxList<TrackChunk>,
+    val beginOffset: Distance,
+    val endOffset: Distance,
+)
+
+data class WithOffset<T>(
+    val value: T,
+    val offset: Distance,
+)
+typealias IdxWithOffset<T> = WithOffset<StaticIdx<T>>
+
+interface PathProperties {
```

Comment:
I'm not sure I understand the API split between PathProperties and TrackProperties; My intuition would be for PathProperties to extend TrackProperties, but it happens to be the other way around